lms/views/admin/application_instance/update.py

```diff
@@ -82,7 +82,10 @@ def update_instance_settings(self):
             value = value.strip() if value else None
 
             if field.format is asbool:
-                value = value == "on"
+                if value == "none":
```

Comment:
I'm not sure if this solves the main issue with the current approach which in my opinion is saving the default option to the DB which makes it difficult to either change the default without a migration or knowing which state changes are explicit vs which ones ares just "saving some other setting".


With this tri-state approach we could `continue` here to avoid changing the DB on this case.

Reply:
> With this tri-state approach we could `continue` here to avoid changing the DB on this case.

If we do that, then changing a setting from an explicit setting back to the default will not work.

What currently happens is that the "Default" option gets serialized as `null`, which de-serializes back to `None` in Python, so the result is the same as if the setting was not present at all.

This could be changed so that the setting is not modified if the current value and new value are both `None`. That would avoid creating redundant entries in the DB when changing other settings. Alternatively this could be handled at a lower layer by not serializing the field at all if the value is `null`.

Reply:
>  Alternatively this could be handled at a lower layer by not serializing the field at all if the value is null.

If we stored this as non-nullable booleans in an hypothetical settings table we wouldn't have to worry to much about this.

> setting from an explicit setting back to the default will not work.

True, I  didn't think about that case, probably because we just abuse this as feature flags but they are in general settings that need intervention from schools to be changed. We only use the "defaultness" for:

- Avoid doing DB migrations for new settings
- Rolling out new features without doing a DB migration


Doing DB migrations is not that bad but if we were going to do that to keep the state I would probably switch to boolean columns as mentioned above.